lambdas/build-cri-oauth-request/src/main/java/uk/gov/di/ipv/core/buildcrioauthrequest/BuildCriOauthRequestHandler.java

```diff
@@ -333,8 +345,41 @@ private JWEObject signEncryptJar(
                         evidenceRequest,
                         context);
 
-        RSAEncrypter rsaEncrypter = new RSAEncrypter(oauthCriConfig.getParsedEncryptionKey());
-        return AuthorizationRequestHelper.createJweObject(rsaEncrypter, signedJWT);
+        RSAKey encKey;
+        if (cachedOAuthCriEncryptionKeys != null
+                && !cachedOAuthCriEncryptionKeys.isEmpty()
+                && cachedOAuthCriEncryptionKeys.get(cri.getId()) != null
+                && !cachedOAuthCriEncryptionKeys.get(cri.getId()).isExpired()) {
+            encKey = cachedOAuthCriEncryptionKeys.get(cri.getId()).key();
+        } else {
+            encKey = oAuthKeyService.getValidEncryptionKey(oauthCriConfig);
+
+            var cacheDuration =
+                    configService.getParameter(ConfigurationVariable.OAUTH_KEY_CACHE_DURATION_MINS)
+                                    == null
```

Comment:
Can/should this ever be null? I think the config service will throw if the config is not found...